src/Parser.ts

```diff
@@ -307,6 +307,45 @@ export function surroundedBy<I, A>(bound: Parser<I, A>): <B>(p: Parser<I, B>) =>
   return between(bound, bound)
 }
 
+/**
+ * Takes a `Parser` and tries to match it without consuming any input.
+ *
+ * @example
+ * import { run } from 'parser-ts/lib/code-frame'
+ * import * as P from 'parser-ts/lib/Parser'
+ * import * as S from 'parser-ts/lib/string'
+ *
+ * const parser = S.fold([
+ *   S.string('hello '),
+ *    P.lookAhead(S.string('world')),
+ *    S.string('wor')
+ * ])
+ *
+ * run(parser, 'hello world')
+ * // { _tag: 'Right', right: 'hello worldwor' }
+ *
+ * @since 0.6.6
+ */
+export const lookAhead = <I, A>(p: Parser<I, A>): Parser<I, A> => i =>
+  E.either.chain(p(i), next => success(next.value, i, i))
```

Comment:
`E.either` is a "mega instance" (i.e. is large) and affects negatively tree shaking, can we use `E.chain` instead?

```ts
pipe(p(i), E.chain(next => success(next.value, i, i)))
```

Reply:
@IMax153 oh never mind, just noticed that there are other occurences of `E.either` here and there...

Reply:
Yea, I noticed this as well when examining the lib in more detail prior to opening the issue that sparked this PR. If desired, I could always open a separate PR that addresses some of the issues with tree shaking. Happy to help with whatever is desired.

Reply:
> If desired, I could always open a separate PR that addresses some of the issues with tree shaking. Happy to help with whatever is desired.

Thanks, it's very kind of yours. But I warn you: that's a pretty boring job :)